miner/worker_test.go

```diff
@@ -0,0 +1,124 @@
+package miner
```

Comment:
These kind of look like integration tests, shouldn't we run them under a build tag and shouldn't they be in a separate package? (miner_test)